### ISSUE TITLE
✍️ Contribution period: Leila Yesufu

### ISSUE THREAD

**Issue by leilayesufu** [ersilia-os/ersilia#820]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [x] Install the Ersilia Model Hub and test the simplest model
- [x] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Install and run an ML model

- [x] Select a model from the suggested list
- [x] Install the model in your system
- [x] Run predictions for the EML
- [x] Compare results with the Ersilia Model Hub implementation!
- [x] Install and run Docker!

### Week 3 - Propose new models

- [x] Suggest a new model and document it (1)
- [x] Suggest a new model and document it (2)
- [x] Suggest a new model and document it (3)

### Week 4 - Prepare your final application

- [x] Submit the final application in the Outreachy website

**Comment by leilayesufu** [comment]
<img width="906" alt="Screenshot 2023-10-02 203111" src="https://github.com/ersilia-os/ersilia/assets/78659709/97d75a64-88b8-41f5-ac52-7a82d35b3973">
Today, 2nd of October 2023.
I got accepted into the outreachy contribution stage and chose to contribute to Ersilia Open Source Initiative project.
I'm very excited to learn and grow during this contribution stage.

**Task 1: Join the communication channels** 
I joined the slack workspace and introduced myself in the general Channel.

**Task 2:  Open a GitHub issue (this one!)**
Opened the issue. Done :)

**Comment by leilayesufu** [comment]
**Task 3:  Install the Ersilia Model Hub and test the simplest model**
**October 2nd 2023**

Following the instructions [here](https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/installation).

I installed the Ersilia Model hub from the Command line Interface.
I'm using a windows device so i had to use the windows subsytem for Linux with an Ubuntu terminal.

There are prerequisites to be installed before downloading the Model hub. This can be seen as follows.
1.   **Install the gcc compiler on Ubuntu terminal**
        This can be down with by running the following
         ` sudo apt-get update`
          `sudo apt install build-essential`
           <img width="388" alt="gcc" src="https://github.com/ersilia-os/ersilia/assets/78659709/55b70dc5-96ae-4f2c-87d0-7739c17bddff">
The gcc compiler has been successfully installed.


2.  **Install  Python and Conda**
     I installed the Miniconda using the following command.
   `mkdir -p ~/miniconda3
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda3/miniconda.sh
bash ~/miniconda3/miniconda.sh -b -u -p ~/miniconda3
rm -rf ~/miniconda3/miniconda.sh
~/miniconda3/bin/conda init bash
~/miniconda3/bin/conda init zsh` 
        <img width="662" alt="Conda" src="https://github.com/ersilia-os/ersilia/assets/78659709/4736b0fe-4de8-4f0a-87fa-e8fe2966d3cd">

 Conda has been successfully installed. This can be tested using `conda --version`

3.  **Install Git and GitHub CLI**

      The next step is to configure Git and GitHub CLI on Conda. This can be done with the following command.
     `conda install gh -c conda-forge` This is to install the github cli
      `gh auth login`  This is to configure your Cli with your credentials.
      Follow the instructions given and you should be logged in to git.

     <img width="525" alt="logged in" src="https://github.com/ersilia-os/ersilia/assets/78659709/41ac63b6-c825-4e60-b378-ab2c4e219809">

4.  **Install and Activate Git LFS**

    This can be done with the following command. 
    `conda install git-lfs -c conda-forge`
     `git-lfs install`

<img width="179" alt="git lfs" src="https://github.com/ersilia-os/ersilia/assets/78659709/6b62b2ec-e786-4f27-9113-3f5cd3668a7c">
   
5. Install The Isaura data lake.
    `#activate ersilia's conda environment (see instructions below to create it)
conda activate ersilia
python -m pip install isaura==0.1`

6. Docker
     I already had docker installed and running on my local machine.
    
<img width="459" alt="docker" src="https://github.com/ersilia-os/ersilia/assets/78659709/b7511aa0-4138-4377-a471-5ca5be56048e">



**Main Installation.**

1.  Install Ersilia
    After all the pre-requisites are met, we are ready to install the Ersilia tool. 
   In your terminal, run the following commands to create and activate the environment
   `conda create -n ersilia python=3.7`
     `conda activate ersilia`

    Then, simply install the Ersilia Python package.
    `
git clone https://github.com/ersilia-os/ersilia.git
cd ersilia
pip install -e .`

     <img width="358" alt="image" src="https://github.com/ersilia-os/ersilia/assets/78659709/3dea1d3b-fd5e-44be-924b-bdac97505a12">

    Erisilia environment has been successfully installed.

Check that the CLI works on your terminal, and explore the available commands
    `ersilia --help`
    
   <img width="516" alt="help" src="https://github.com/ersilia-os/ersilia/assets/78659709/58f82a27-46d7-4a49-81c1-322ad76c37cf">

   `ersilia catalog`
     <img width="713" alt="catalog" src="https://github.com/ersilia-os/ersilia/assets/78659709/561b5a7d-b3be-40b2-8da6-6ac122dcc69c">


We have successfully installed the Ersilia model hub.

**Comment by leilayesufu** [comment]
Task 3: Install the Ersilia Model Hub and test the simplest model
This is a continuation, in this part we're going to be testing a very simple model with Ersilia .
Run the following commands.

`ersilia -v fetch eos3b5e`

`ersilia serve eos3b5e`

The first two ran without errors

<img width="478" alt="serve" src="https://github.com/ersilia-os/ersilia/assets/78659709/f9bda7f3-b3ef-4017-b6ba-e839e8109e8a">

The calculate however gave me some errors

`ersilia -v api calculate -i "CCCC"`

I tried it at first and got an error such as [this](https://github.com/ersilia-os/ersilia/issues/821#issuecomment-1743901761)
Upon further inspection of the logs, i noticed that the key error at the end of the traceback was due to the fact that the code was trying to access a dictionary key 'calculate' in the schema, but was not finding it.

Upon double checking the schema file at /home/leila/eos/dest/eos3b5e/api_schema.json i confirmed that there was no calculate key only a run key.  I then updated the schema as thus using values from the logs.

<img width="572" alt="commands" src="https://github.com/ersilia-os/ersilia/assets/78659709/384b502d-7b57-4173-99c5-336b653ba77b">

I then ran it again and got another error but this wasn't due to key error, it was due to an issue with reading input columns in the code cause the object had no len

<img width="653" alt="run error" src="https://github.com/ersilia-os/ersilia/assets/78659709/765137a5-ba5c-4109-aefd-9227af2a6688">

Opening the code at /home/leila/ersilia/ersilia/io/readers/file.py, In the read_input_columns function at line 321.
I changed the code from 
`if len(h) == 1:` to
`if h is not None and len(h) == 1:`

This change ensures that the length check is only performed when h is not None. If h is None,the code proceeds accordingly.
After making these changes, i ran the command 

`ersilia -v api calculate -i "CCCC"`

and i got this

<img width="667" alt="output null" src="https://github.com/ersilia-os/ersilia/assets/78659709/cfbffa90-ffb4-46d6-a22e-5197fff2a9cd">

I have gotten the input section but not the output. upon further troubleshooting i found out my mistake in the schema file at /home/leila/eos/dest/eos3b5e/api_schema.json when creating the calculate i used "result" instead of "mw"
After changing that i get

<img width="452" alt="nulk" src="https://github.com/ersilia-os/ersilia/assets/78659709/e843eecb-350d-41e1-829a-1d84197138d1">

I'm still finding it difficult to get the output value

**Comment by HellenNamulinda** [comment]
Hello @leilayesufu,
You don't have to change the schema as it is generated automatically when you fetch and serve the model.
Instead of using calculate as the api, use run. 
Forexample, instead of `ersilia -v api calculate -i "CCCC"`, use `ersilia -v run -i "CCCC"`

**Comment by leilayesufu** [comment]
Seen, Thank you very much

**Comment by leilayesufu** [comment]
After a correction by @HellenNamulinda 
She informed me that i could have used 
`ersilia -v api run -i "CCCC"` instead of adding my own key to the schema and trying `ersilia -v api calculate -i "CCCC"`

After running `ersilia -v run -i "CCCC"`
I got the output.

<img width="400" alt="output" src="https://github.com/ersilia-os/ersilia/assets/78659709/dd5a0694-a537-42a8-a3cc-985443115cfa">

Task 3 has been completed successfully. Thank you @HellenNamulinda

**Comment by leilayesufu** [comment]
**Task 4: Write a motivation statement to work at Ersilia**
October 3rd 2023

I am thrilled at the prospect of joining Ersilia, After reading about the history, mission, vision and goals of Ersilia.
I was drawn to Ersilia's Open Source Initiative for its profound commitment to driving positive change in healthcare accessibility and its innovative approach to bridging the gap between AI/ML expertise and biomedical research.

Ersilia's Open Source Initiative resonates with me on a deeply personal level, as it reflects my own commitment to leveraging tools and technology for positive change in healthcare accessibility. Hailing from Nigeria, a middle-income country, I've witnessed firsthand the challenges that communities face in accessing cutting-edge tools for biomedical research.

I really appreciate the value that this project can bring up to the world and i would love to work towards bringing that value.
I am also enthusiastic about the opportunity to join a team that actively promotes inclusivity and values the collective contribution of employees, contractors, interns, and volunteers.

In conclusion, I am excited about the possibility of contributing to Ersilia's mission and vision, and I am eager to bring my skills and dedication to a team that is at the forefront of making a positive impact on global healthcare.

**Comment by leilayesufu** [comment]
**Task 5:  Submit your first contribution to the Outreachy site**

I just submitted my first Contribution to the Outreachy site : ))

**Comment by leilayesufu** [comment]
**Week 2 - Install and run an ML model**
**Task 1: Select a model from the suggested list**

October 3rd 2023

After going through the proposed models, I am particularly drawn to the NCATS Rat Liver Microsomal Stability [here.](https://github.com/ncats/ncats-adme) . I chose this model because of the alignment of my interests and the potential impact on drug development and healthcare. The specific model focuses on predicting ADME(Absorption, Distribution, Metabolism, and Excretion) properties related to liver microsomes especially Rat Liver Microsomal Stability. It presents an opportunity to make meaningful contributions to drug development and advance my understanding of machine learning in the biomedical context.
Understanding how drugs interact with the liver microsomes is crucial in drug development, making this Model more appealing to me.

**Comment by DhanshreeA** [comment]
Hi @leilayesufu, great work! Thanks for the updates. :)

**Comment by leilayesufu** [comment]
Thank youuu! @DhanshreeA

**Comment by leilayesufu** [comment]
Good day, For everyone making reference to here.
According to mentors the base code should not be changed.
[Kindly refer here](https://github.com/ersilia-os/ersilia/issues/822#issuecomment-1745812897)

**Comment by leilayesufu** [comment]
@DhanshreeA I've successfully completed Week 1 - Get to know the community, Do i go ahead to Week 2 - Install and run an ML model?

**Comment by DhanshreeA** [comment]
Hi @leilayesufu yes absolutely. Thanks for the updates. You can go ahead and get started with week 2 tasks.

**Comment by leilayesufu** [comment]
## Week 2 - Install and run an ML model

**Task 6: Select a model from the suggested list**

After going through the proposed models, i selected NCATS Rat Liver Microsomal Stability: https://github.com/ncats/ncats-adme

After reading this publication for the model https://pubs.acs.org/doi/10.1021/acs.jcim.2c00297
I learnt that the objective is to predict the human plasma protein binding (PPB *) property of compounds with the use of an interpretable deep learning method.

This sparked my interest because the use of interpretable deep learning method the model will not only provide predictions but it will also provide understanding into factors that influence PPB.
I am fascinated at the potential impact on drug development, and the scientific insights that can be achieved from these predictions, I am also curious to know to if these drugs once they get inside our bodies, will stick to plasma proteins or move freely.

**Comment by DhanshreeA** [comment]
> ## Week 2 - Install and run an ML model
> 
> **Task 6: Select a model from the suggested list**
> 
> After going through the proposed models, i selected NCATS Rat Liver Microsomal Stability: https://github.com/ncats/ncats-adme
> 
> After reading this publication for the model https://pubs.acs.org/doi/10.1021/acs.jcim.2c00297 I learnt that the objective is to predict the human plasma protein binding (PPB *) property of compounds with the use of an interpretable deep learning method.
> 
> This sparked my interest because the use of interpretable deep learning method the model will not only provide predictions but it will also provide understanding into factors that influence PPB. I am fascinated at the potential impact on drug development, and the scientific insights that can be achieved from these predictions, I am also curious to know to if these drugs once they get inside our bodies, will stick to plasma proteins or move freely.

Hi @leilayesufu thanks for the updates. However there's a slight confusion in your comment. You mention selecting "NCATS Rat Liver Microsomal Stability", however it appears you have worked with "Plasma Protein Binding (IDL-PPBopt)" instead. These are two different models.

**Comment by leilayesufu** [comment]
Ah, Thank you so much for the correction!!!
I'm working on Plasma Protein Binding (IDL-PPBopt): https://github.com/Louchaofeng/IDL-PPBopt

I'll update it now. Thank you for the information on scikit-learn as well

**Comment by leilayesufu** [comment]
To test the IDL-PPBopt model, I located ipynb jupyter notebook that is used run the code seen [here](https://github.com/Louchaofeng/IDL-PPBopt/blob/main/Code/IDL-PPBopt.ipynb) , i then extracted all the steps needed to predict values and saved it in a python file i created and named model.py, I did this because i want to use python for the predictions.

The code was still dependent on cuda and gpu, so i had to edit the model.py file i created.

I first defined the device to CPU using the following command

`device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')`

I then edited the all the files dependent on cuda

to load the file, i changed the loading command from `model.cuda()` to ` model.to(device)`

Then i tried running the file, i got the error shown [here](https://github.com/ersilia-os/ersilia/files/12828324/Log2.txt), This was because the files made use of imports that is dependent on IPython from jupyter. So i had to comment out all the lines dependent on IPython.

After this was done, I tried running it again and i got the error shown in the log [here](https://github.com/ersilia-os/ersilia/files/12828308/Log1.txt)

To fix this, i changed the line of code

`best_model = torch.load('saved_models/model_ppb_3922_Tue_Dec_22_22-23-22_2020_'+'54'+'.pt')`  to 

`best_model = torch.load('saved_models/model_ppb_3922_Tue_Dec_22_22-23-22_2020_'+'54'+'.pt',map_location=torch.device('cpu'))`

Then i ran it again and it finally worked.

view log file [here](https://github.com/ersilia-os/ersilia/files/12828377/homeleilaminiconda3envsIDL-PPBoptli.txt).

View the CSV output of the default input file in the [Plasma Protein Binding (IDL-PPBopt)](https://github.com/Louchaofeng/IDL-PPBopt/blob/main/Code/IDL-PPBopt.ipynb) model here
[LeilaYesufuPredictions.csv](https://github.com/ersilia-os/ersilia/files/12828400/LeilaYesufuPredictions.csv)


The IDL-PPBopt model has been installed and run on my system.
Task 7:  Install the model in your system has been completed successfully

**Comment by leilayesufu** [comment]
To Run predictions for the Essential Medicines List with the IDL-PPBopt model.

I inputted the given EML list into my model.py i editted the heading from smiles to cano_smiles then i ran the command `python model.py`

This is the successful log, and this is the predictions output
[log for IDL-PPBopt.txt](https://github.com/ersilia-os/ersilia/files/12830981/log4.txt)

[IDL-PPBopt model_eml_canonical_output.csv](https://github.com/ersilia-os/ersilia/files/12830975/IDL-PPBopt.model_eml_canonical_output.csv)



**Explanation of the result**

The model uses an Interpretable Deep Learning Method helps us understand how likely a compound is to stick to plasma proteins, 
The model takes CANONALIZED SMILES as input and returns PPB fractions as output.
PPB fraction above 80% represents a good affinity or connection with plasma proteins
While PPB fraction less than 40% presents a low level of affinity.

**Comment by leilayesufu** [comment]
## Task 4: understand Ersilia's backend and running the EML list with Ersilia

> Ersilia can run by downloading models from GitHub (using Git-LFS), from S3 buckets (our AWS backend) and by downloading models as Docker containers.

I decided to challenge myself and run the model with Docker containers 

I located the model in the in the Ersilia Model Hub. Then i went to docker hub and got the command for pulling the image.

"docker pull ersiliaos/eos22io" view the log [here](https://github.com/ersilia-os/ersilia/files/12837639/pull.txt)

I ran this in a detached mode as seen .
<img width="743" alt="container" src="https://github.com/ersilia-os/ersilia/assets/78659709/3f5638a8-cfa9-4d2c-81b1-1e65b1d66d13">

After running the container, i inspected it [here](https://github.com/ersilia-os/ersilia/files/12837643/docker_inspect.txt) and found that the container has a custom entry point specified, and it's using the shell sh

so i accessed the shell terminal with the following command `docker exec -it containerid sh`and i ran the predictions of the EML file given on it

here are the [logs](https://github.com/ersilia-os/ersilia/files/12837649/ersilia_comparison.txt)

**explanation of logs** 

i ran this two commands as seen in the Dockerfile
`cd /root/`
`sh docker-entrypoint.sh`

The model has already been served then i used this command to download the EML list into the Docker shell

`wget https://raw.githubusercontent.com/ersilia-os/ersilia/master/notebooks/eml_canonical.csv`

then i ran this command to run the predictions on the downloaded `ersilia -v api run -i eml_canonical.csv -o ersilia_output.csv`

to run the file and save the result in a new file called ersilia_output.csv

The file can be seen here.

[ersilia_output.csv](https://github.com/ersilia-os/ersilia/files/12837658/ersilia_output.csv)


I exited the docker interactive shell and used the below command to copy the output generated from inside the docker container to my home directory

 `docker cp dbff41b896d1:/root/ersilia_output.csv /home/leila/ersilia/log_files`

## Comparison of the ouput from IDL-PPBopt and using ersilia to run it

The IDL-PPBopt model with EML: [IDL-PPBopt.model_eml_canonical_output.csv](https://github.com/ersilia-os/ersilia/files/12837679/IDL-PPBopt.model_eml_canonical_output.csv)
Running EML with Ersilia/ersiliaos/eos22io: [ersilia_output.csv](https://github.com/ersilia-os/ersilia/files/12837658/ersilia_output.csv)


I checked the output of every single result and they were the same apart from number 97, 98 in the IDL-PPBopt model  which had an output of 0.93916506, 0.9521944 while in the ersilia one had an output of 0.9391651,0.95219433 so a slight rounding up difference.

It should also be noted that the running with the ersilia model produced 9 nan results making it 442 in number while the IDL-PPBopt model result  left the nan results out making it 433 in number.

This is because as seen in the log of the IDL-PPBopt model [here](https://github.com/ersilia-os/ersilia/files/12837748/log4.txt), 9 compounds could not be featured.

From my understanding, the ersiliaos/eos22io featured them and gave a nan result while IDL-PPBopt model left them out completely


I have successfully completed week 2 tasks.
 

- [ ] Compare results with the Ersilia Model Hub implementation!
- [ ] Install and run Docker!

**Comment by DhanshreeA** [comment]
Hi @leilayesufu thank you for the detailed updates. Proteins are organic compounds (containing C-H or C-C bonds), whereas the nine compounds that are not featurized within the original code are inorganic compounds or salts. Since the input and output file for a model within ersilia needs to be of the same length, ersilia simply outputs null corresponding to the molecules that don't produce an output (because they don't get featurized)

**Comment by leilayesufu** [comment]
Thank you very much!

**Comment by leilayesufu** [comment]
## WEEK 3

### Suggest a new model and document it (1)


Slug: CLAMP

Model title: Enhancing Activity Prediction Models in Drug Discovery with the Ability to Understand Human Language

Publication: https://arxiv.org/abs/2303.03363

Github repo: https://github.com/ml-jku/clamp#clamp-clamp

license: GPLv3

Code: Python

Year: 2023

Description:

CLAMP (Contrastive Language-Assay Molecule Pre-Training) is a ML model trained on pairs molecule-bioassay pairs. It can be communicated to in natural language. CLAMP can be used to make predictions about which molecules are most relevant to a particular bioassay when given a textual description of the bioassay.  
From the publication, my understanding is that CLAMP can be used to predict how well a particular molecule might work for a given task, like being a drug to treat a disease or having specific properties for a material. It does this by understanding textual descriptions of tasks. you can give CLAMP a list of molecules (SMILES) and a text or description and it predicts which is the best molecule according to the description. The higher the probability means that the model believes that the molecule is suitable for the written bioassay.

Relevance to Ersilia: I believe it'll be relevant to Ersilia because it is an AI/ML tool that can be used in drug discovery.

**Comment by leilayesufu** [comment]
### Suggest a new model and document it (2)

Slug: DrugApp

Title: Machine learning-based prediction of drug approvals using molecular, physicochemical, clinical trial, and patent-related features

Publication: https://www.tandfonline.com/doi/abs/10.1080/17460441.2023.2153830

Source Code: https://github.com/fulyaciray/DrugApp

Description: DrugApp is a tool that uses machine learning and different data sources to predict the likelihood/potential of regulatory approval for drugs. 
The tool uses data from clinical trials and patent-related information. It also considers the physical and chemical properties of the drug candidate molecules. The prediction method used is random forest classifier to build disease group-specific approval prediction models.
The project contributes to drug discovery as it helps the scientists/ drug makers predict if their drug will be approved by regulatory bodies, It also saves time and resources in the drug discovery process.

License: GPL-3.0 license

Code: Python

Year: 2022

**Comment by leilayesufu** [comment]
Slug: RLBind

Title: RLBind: a deep learning method to predict RNA–ligand binding sites

Publication: https://www.researchgate.net/publication/365587894_RLBind_a_deep_learning_method_to_predict_RNA-ligand_binding_sites

Source Code: https://github.com/KailiWang1/RLBind/tree/main

Description: This model uses a deep convolutional neural network (CNN) to predict locations on RNA molecules where ligands are most likely to bind. This can be useful when designing drugs and in drug discovery and also in using RNAs as targets for treating diseases.
RNA according to google means ribonucleic acid, a nucleic acid present in all living cells.

License:  [Apache-2.0 license](https://github.com/KailiWang1/RLBind/blob/main/LICENSE)

Code: Python

Year:

**Comment by leilayesufu** [comment]
## Running the models.

Model 1: [CLAMP]( https://github.com/ml-jku/clamp#clamp-clamp)

I set up the environment using the commands on the GitHub repository, a  yaml file was given with all the dependencies so running the `conda env create -f env.yml`  created the environment and installed all the python and pip dependencies.
I activated the environment .
I then added the pretrained model given in a main.py file and ran `python main.py`

I encountered the following error [cliperror.txt](https://github.com/ersilia-os/ersilia/files/12938148/cliperror.txt), This is due to clip being a dependency not mentioned in the environment file, this was fixed by installing (Contrastive Language-Image Pretraining) model via 
`pip install git+https://github.com/openai/CLIP.git`

then i ran the file again and i got the following required output here [output.txt](https://github.com/ersilia-os/ersilia/files/12938154/output.txt).

I then changed the content of the pretrained model to have the first 4 smiles of the EML file with the same bioassay and got this result here. 
[smiles.txt](https://github.com/ersilia-os/ersilia/files/12938161/smiles.txt)

- [ ] The clamp model has been executed and tested locally on my system.

**Comment by leilayesufu** [comment]
- ## Model 3: [RLBind](https://github.com/KailiWang1/RLBind/tree/main)

For this model, i cloned the git repository and i installed the listed requirements one after the other. I didn't use the environment yaml file given because it is for a gpu enabled version.

`conda create -n RLBind python==3.7` to create a new environment
I installed 
    `conda install pytorch==1.5.0 torchvision cpuonly -c pytorch` pytorch cpu only 
     `pip install --upgrade torch`
     `pip install numpy==1.16.4` numpy
      `pip install scikit-learn` scikit-learn
       `pip install pandas`

To test the model i ran 
`cd ./src/`
`python predict.py`

I got the following error [cuda.txt](https://github.com/ersilia-os/ersilia/files/12961692/cuda.txt) due to the use of cuda in the file, so i edited the predict.py file so it wouldn't depend on cuda.
I removed the line `model = model.cuda()` and changed the line from 
`model.load_state_dict(torch.load(model_file))` to `model.load_state_dict(torch.load(model_file,map_location=torch.device('cpu')))`

then i ran `python predict.py` again and i got the predictions as seen below.

[prediction.txt](https://github.com/ersilia-os/ersilia/files/12961781/prediction.txt)

**Comment by leilayesufu** [comment]
@DhanshreeA HI, please i'd love some feedback on the models i provided

**Comment by GemmaTuron** [comment]
Hello,

Thanks for your work during the Outreachy contribution period, we hope you enjoyed it! We will now close this issue while we work on the selection of interns. Thanks again!
